src/main/java/microsoft/exchange/webservices/data/core/ExchangeServiceBase.java

```diff
@@ -820,6 +805,48 @@ public void setWebProxy(WebProxy value) {
   public Map<String, String> getHttpHeaders() {
     return this.httpHeaders;
   }
+  
+  protected final CloseableHttpClient getHttpClient(){
```

Comment:
Sure, I can do that. But wouldn't it be better to do that in another pull request?

I'm happy to do it here, but to me, it seems like this pull request should solve one issue (thread safety) and not expand to other issues (like this one, which has already been reported separately as #423). Let me know.